client/ayon_maya/api/lib.py

```diff
@@ -4241,3 +4242,75 @@ def get_node_index_under_parent(node: str) -> int:
         return cmds.listRelatives(parent,
                                   children=True,
                                   fullPath=True).index(node)
+
+
+def search_textures(filepath):
+    """Search all texture files on disk.
+
+    This also parses to full sequences for those with dynamic patterns
+    like <UDIM> and %04d in the filename.
+
+    Args:
+        filepath (str): The full path to the file, including any
+            dynamic patterns like <UDIM> or %04d
+
+    Returns:
+        list: The files found on disk
+
+    """
+    filename = os.path.basename(filepath)
+
+    # Collect full sequence if it matches a sequence pattern
+    if len(filename.split(".")) > 2:
+
+        # For UDIM based textures (tiles)
+        if "<UDIM>" in filename:
+            sequences = self.get_sequence(filepath,
+                                          pattern="<UDIM>")
+            if sequences:
+                return sequences
+
+        # Frame/time - Based textures (animated masks f.e)
+        elif "%04d" in filename:
+            sequences = self.get_sequence(filepath,
+                                          pattern="%04d")
+            if sequences:
+                return sequences
+
+    # Assuming it is a fixed name (single file)
+    if os.path.exists(filepath):
+        return [filepath]
+
+    return []
+
+
+def get_sequence(filepath, pattern="%04d"):
+    """Get sequence from filename.
+
+    This will only return files if they exist on disk as it tries
+    to collect the sequence using the filename pattern and searching
+    for them on disk.
+
+    Supports negative frame ranges like -001, 0000, 0001 and -0001,
+    0000, 0001.
+
+    Arguments:
+        filepath (str): The full path to filename containing the given
+        pattern.
+        pattern (str): The pattern to swap with the variable frame number.
+
+    Returns:
+        list: file sequence.
+
+    """
+    escaped = re.escape(filepath)
+    re_pattern = escaped.replace(pattern, "-?[0-9]+")
+
+    source_dir = os.path.dirname(filepath)
+    files = [f for f in os.listdir(source_dir)
+                if re.match(re_pattern, f)]
+
+    pattern = [clique.PATTERNS["frames"]]
+    collection, remainder = clique.assemble(files, patterns=pattern)
```

Comment:
This will fail on 'sequences' that are a single file. It needs the `minimum_items=1` argument or whatever clique needs.

Also, this returns a `list[clique.Collection]` which I think may be good to put in the type hints, because it is **not** returning a list of filenames.

Reply:
Guess we have fixed in other PR before the addon separation, I would update the function with that code.